buildSrc/src/main/java/org/opensearch/gradle/OpenSearchTestBasePlugin.java

```diff
@@ -109,7 +109,11 @@ public void execute(Task t) {
                         test.systemProperty("java.locale.providers", "SPI,JRE");
                     } else {
                         test.systemProperty("java.locale.providers", "SPI,COMPAT");
-                        test.jvmArgs("--illegal-access=warn", "-Djava.security.manager=allow");
+                        test.jvmArgs("--illegal-access=warn");
+                    }
+                    if (BuildParams.getIsRuntimeJavaHomeSet() == false
```

Comment:
Please use `test.getJavaVersion()` since it could use another JDK (fe toolchain etc)

Reply:
do you mean 
`if(test.getJavaVersion().compareTo(JavaVersion.VERSION_17) > 0)) {}` ?

Reply:
Something along these lines, but should not be the `BuildParams`